packages/ethers/src/__tests__/provider-adapter.test.ts

```diff
@@ -29,6 +31,29 @@ describe("Simple Account Tests", async () => {
       "0xbfe07c95623df55ae939ddf4757563286472ef8c0ebe4b84d5e774a653b7eb67735cb5b63d15bb18510d64a97e6e3001a5f9818f89f2f7f076e559248a7ccf7d1c"
     );
   });
+
+  it("should correctly do runtime validation when entryPoint is invalid", async () => {
+    expect(
+      () =>
+        new EthersProviderAdapter({
+          accountProvider: new SmartAccountProvider({
+            rpcProvider: alchemyProvider.connection.url,
+            chain: 1 as unknown as Chain,
```

Comment:
why are we doing this instead of getting chain from viem?

Reply:
this forces compile time to check, but fail at runtime, which this test is checking again

Reply:
so if we import `mainnet` from viem/chains and use here instead of 1 casted to Chain, it fails runtime?

Reply:
no mainnet would pass b/c it's the same type as expected, where we want this to fail. it's mocking for instance if someone specific their chain via an .env variable, which some ts environments would read as any and perhaps accept whereas they shouldn't.

if we use `mainnet` we get:
![Screenshot 2023-11-02 at 3 57 05 PM](https://github.com/alchemyplatform/aa-sdk/assets/43521356/81d2eb16-479a-436e-936e-fb2da14bd02a)